[#98748824] Add support for version pinning the archive and gandalf servers

actionjack opened the pull request:
[Pin the versions of components that we use](https://www.pivotaltracker.com/story/show/98748824)

**What**

To prevent breaking changes of components (e.g. Tsuru API, MongoDB, etc) causing the platform to break we should pin or build to a particular version, this will ensure a consistent build.

This PR adds version control for the [archive server](https://github.com/tsuru/archive-server) and the [gandalf api](https://github.com/tsuru/gandalf)

**How to review this PR**

This PR has been written to be reviewed in the following context:
- I want to create a vagrant environment to help me develop and test locally
- I want to parameterise some of the packages so I would like to move them into the `task/main.yml` file so they are easier to find and modify
- I want to specifically allow the `archive server` and the `gandalf api` to be pinned to a specific version
- I want to update the documentation so others know how to pin a specific version of the `archive server` and the `gandalf api`

**How to test this PR**

A vagrant box has been provided for testing.

```
$ vagrant up
```

You can supply the following snippet to the `site.yml` for testing version pinning:

```
vars:
  - archive_server_version: 0.1.2-0~trusty1
  - gandalf_server_version: 0.7.0-0~trusty1
```

**Who should review this PR**
- Any one in the world, man, woman, child or chicken!

actionjack committed:
Create vagrant enviroment for local development

actionjack committed:
Move packages to be installed out of variables

I need to pin specific versions of these packages so it makes sense to
move them out of `vars/main.yml` into `tasks/main.yml` first.

dcarley commented on tasks/main.yml:
```diff
@@ -4,9 +4,27 @@
+
+- name: Install a specific version of Archive Server
+  apt: name=archive-server={{ archive_server_version }}
+  when: archive_server_version is defined
```
Splitting these out to separate tasks is going to increase the installation time. Is it possible to provide versions through `with_items`?

actionjack replied:
I didn't perceive it to take that much longer, but that is purely subjective. It could possibly make the logic more complicated and difficult to follow but e.g. breaking out into python in the with_items e.g.

`{% if archive_server_version is defined %}archive-server={{ archive_server_version }}{% elif %}archive-server{% endif %}`

What do you think?

dcarley replied:
We talked about this offline and figured it was better to change. Martin did so.

actionjack committed:
Allow gandalf and archive servers to be pinned

Allow gandalf server and archive server to be pinned to a specific
package version.

actionjack committed:
Update README.md documentation

Update README.md documentation on the use of `archive_server_version`
and `gandalf_server_version`

dcarley commented:
This LGTM but I haven't had a chance to re-test it. Out tomorrow, so un-assigning myself. Can someone else pick it up?

keymon commented:
:+1: for me. Works defining and not defining versions.
